app/src/main/java/com/android/sample/ui/focus/FocusModeManager.kt

```
@@ -0,0 +1,60 @@
+package com.android.sample.ui.focus
+
+import android.Manifest
+import android.content.Context
+import android.media.RingtoneManager
+import android.os.VibrationEffect
+import android.os.Vibrator
+import androidx.annotation.RequiresPermission
+import androidx.work.WorkManager
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+object FocusModeManager {
+
+  /**
+   * Activate Focus Mode:
+   * - Cancels all scheduled notifications (tag = "notifications")
+   * - Gives short haptic feedback
+   */
+  fun activate(context: Context) {
+    CoroutineScope(Dispatchers.Default).launch {
+      try {
+        WorkManager.getInstance(context).cancelAllWorkByTag("notifications")
+        giveHapticFeedback(context)
+      } catch (e: Exception) {}
```

We shouldn’t swallow all exceptions silently here. At minimum, log them (Log.w("FocusModeManager", "Failed to activate focus mode", e)) or propagate via a result type. Silent failures will make it hard to debug notification/cancellation issues in production.